portfolio/src/main/webapp/script.js

```diff
@@ -42,16 +42,15 @@ function initTic(){
     checkGrid[i] = [];
```

Comment:
Another nit: you're initializing checkGrid[i] as an empty array [], but it is meant to be storing characters (either 'X' or 'O'). In that case it would be better to initialize it as an empty character to denote how it will be used.

However the code is actually using it as a boolean, in line 57/58:
```
    if(checkGrid[i][j]!='X' && checkGrid[i][j]!='O') {
      // ...
    }
```

you could directly use a boolean, e.g. `if (!gridChecked[i][j])`, or even better, you could drop the separate variable altogether, and add this property to boxGrid[i][j], the same way you're adding myRow and myColumn. This way it's more consistent with how you handle rows and columns. That way you could do:

```
    boxGrid[i][j].myRow=i;  
    boxGrid[i][j].myColumn=j;  
    boxGrid[i][j].checked = false;  
```

and in the play() function:

```
    var i = this.myRow;  
    var j = this.myColumn;  
    var checked = this.checked;  
    if (!checked) {  
       // ...  
    }
```